cmake/OpenCVInstallLayout.cmake

```diff
@@ -0,0 +1,122 @@
+# Following macro updates multiple variables at once:
+#  - OPENCV_BIN_INSTALL_PATH
+#  - OPENCV_LIB_INSTALL_PATH
+#  - OPENCV_CONFIG_INSTALL_PATH
+#  - ...
+macro(ocv_set_install_layout)
+  unset(_var)
+  foreach(arg ${ARGN})
+    if(NOT _var)
+      set(_var ${arg})
+    else()
+      ocv_update(OPENCV_${_var}_INSTALL_PATH ${arg})
+      unset(_var)
+    endif()
+  endforeach()
+endmacro()
+
+if(OPENCV_CUSTOM_INSTALL_LAYOUT)
+
+  include("${OPENCV_CUSTOM_INSTALL_LAYOUT}")
+
+elseif(ANDROID)
+
+  ocv_set_install_layout(
+    BIN            "sdk/native/bin/${ANDROID_NDK_ABI_NAME}"
+    TEST           "sdk/native/bin/${ANDROID_NDK_ABI_NAME}"
+    SAMPLES_BIN    "sdk/native/samples/${ANDROID_NDK_ABI_NAME}"
+    LIB            "sdk/native/libs/${ANDROID_NDK_ABI_NAME}"
+    LIB_ARCHIVE    "sdk/native/staticlibs/${ANDROID_NDK_ABI_NAME}"
+    3P_LIB         "sdk/native/3rdparty/libs/${ANDROID_NDK_ABI_NAME}"
+    CONFIG         "sdk/native/jni"
+    INCLUDE        "sdk/native/jni/include"
+    SAMPLES_SRC    "samples/native"
+    LICENSES       "sdk/etc/licenses"
+    DOC            "doc"
+    TEST_DATA      "sdk/etc/testdata"
+    OTHER          "sdk/etc"
+    JAR            "."
+    JNI            "sdk/native/libs/${ANDROID_NDK_ABI_NAME}"
+    JNI_BIN        "sdk/native/libs/${ANDROID_NDK_ABI_NAME}"
+  )
+
+elseif(WIN32 AND CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
+
+  if(DEFINED OpenCV_RUNTIME AND DEFINED OpenCV_ARCH)
+    set(_prefix "${OpenCV_ARCH}/${OpenCV_RUNTIME}/")
+  else()
+    message(STATUS "Can't detect runtime and/or arch")
+    set(_prefix "")
+  endif()
+  if(OpenCV_STATIC)
+    set(_suffix "staticlib")
+  else()
+    set(_suffix "lib")
+  endif()
+  if(INSTALL_CREATE_DISTRIB)
+    set(_jni_suffix "/${OpenCV_ARCH}")
+  else()
+    set(_jni_suffix "")
+  endif()
+
+  ocv_set_install_layout(
+    BIN           "${_prefix}bin"
+    TEST          "${_prefix}bin"
+    SAMPLES_BIN   "${_prefix}samples"
+    LIB           "${_prefix}${_suffix}"
+    LIB_ARCHIVE   "${_prefix}${_suffix}"
+    3P_LIB        "${_prefix}${_suffix}"
+    CONFIG        "."
+    INCLUDE       "include"
+    SAMPLES_SRC   "samples"
+    LICENSES      "etc/licenses"
```

Comment:
I would prefer to keep original expressions and put in after dependent item ("OTHER") in this list:
```
${OPENCV_OTHER_INSTALL_PATH}/licenses
```

All cases (from 3.4 branch):
```
$ grep -n -E 'INSTALL_PATH.*\".*OPENCV_.*INSTALL_PATH' CMakeLists.txt 
566:  ocv_update(OPENCV_TEST_INSTALL_PATH "${OPENCV_BIN_INSTALL_PATH}")
591:  ocv_update(OPENCV_LICENSES_INSTALL_PATH "${OPENCV_OTHER_INSTALL_PATH}/licenses")
607:    ocv_update(OPENCV_LICENSES_INSTALL_PATH "${OPENCV_OTHER_INSTALL_PATH}/licenses")
650:  ocv_update(OPENCV_JNI_INSTALL_PATH "${OPENCV_LIB_INSTALL_PATH}")
652:  ocv_update(OPENCV_JNI_INSTALL_PATH "${OPENCV_JAR_INSTALL_PATH}/${OpenCV_ARCH}")
654:  ocv_update(OPENCV_JNI_INSTALL_PATH "${OPENCV_JAR_INSTALL_PATH}")
656:ocv_update(OPENCV_JNI_BIN_INSTALL_PATH "${OPENCV_JNI_INSTALL_PATH}")
```